.github/workflows/test.yml

```diff
@@ -9,18 +9,38 @@ on:
 jobs:
   test:
     runs-on: ubuntu-latest
-
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+
       - name: Use Node.js
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - uses: pnpm/action-setup@v2.0.1
+        name: Install pnpm
+        id: pnpm-install
         with:
-          node-version: '14'
-          cache: 'yarn'
-      - run: yarn install
-      - run: yarn types:check
-      - run: yarn lint
-      - run: yarn build
-      - run: yarn test
+          version: 7
+          run_install: false
+
+      - name: Get pnpm store directory
+        id: pnpm-cache
+        run: |
+          echo "::set-output name=pnpm_cache_dir::$(pnpm store path)"
+
+      - uses: actions/cache@v3
+        name: Setup pnpm cache
+        with:
+          path: ${{ steps.pnpm-cache.outputs.pnpm_cache_dir }}
```

Comment:
From turborepo discord

> The problem with using action/cache is the fact that you most specify a restore key which means it will keep only one cache per build.
When you are using turborepo, this means that you have a mono-repo, which probably will produce multiple artifacts. 
Think about a monorepo with A,B,C,D artifacts.
The first build will miss all, create a cache to all (the restore key will be somehow generated...) in one cache.
Then you changed only A, this build will miss, since, the previous cache contains A, and it should be missed. = no gain.